### PR TITLE
Add ModifyBlockEvent

### PR DESCRIPTION
**Pull request by LatvianModder**
This event is needed to resolve all custom mod integrations. Instead mods would fire this event, and protection mods could catch and cancel it. Examples: Gravity Gun picking up block, Chisels&Bits adding/removing bit from the block (there even is an event in this mod's api - https://github.com/AlgorithmX2/Chisels-and-Bits/blob/1.10/src/main/java/mod/chiselsandbits/api/EventBlockBitModification.java), wrenches rotating / removing blocks, etc.
The problem with APIs is that you either have to know all mods and all of their APIs (if they have one at all) or wait for someone to request support.

### CODE REVIEW

**Comment by williewillus** [comment]
I don't think this belongs in PIE.
PIE strictly revolves against all the complex calls that issue from the item and block when you do clicking.

In addition, PIE is already getting crowded as it is.

This should just be moved to a new top level class

**Comment by williewillus** [comment]
And the doc is vague what is "nonstandard"?

When I click a wrench on a block several PIE's are already fired.

**Comment by LatvianModder** [comment]
Am I allowed to add real examples in description? Like Portal Gun, Gravity Gun, C&B etc

**Comment by williewillus** [comment]
That still doesn't answer the question that the other pie's are already sufficient to use for permissions 0.o

Pending PR#3070, you can detect and cancel any interaction on anything with whatever result you want. If you want permission checking on c+b for example you check for a left click block on a c+b block, etc.

**Comment by blay09** [comment]
EE's original philosophers stone comes to mind, where a right-click on a single block (=> a single PIE) affects multiple blocks when turning cobble into stone for example.

**Comment by williewillus** [comment]
Those should be break and place events

**Comment by williewillus** [comment]
ProjectE does that, our phil stone fires break and place events for blocks it touches.

So if anything this should go with those events, not this one

**Comment by LatvianModder** [comment]
It's not a PIE.. It extends PlayerEvent

File: src/main/java/net/minecraftforge/event/world/ModifyBlockEvent.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    {
+        return action;
+    }
+
+    /**
+     * @return Convenience method to get the block state.
+     */
+    public IBlockState getBlockState()
+    {
+        return getWorld().getGroundAboveSeaLevel(getPos());
```
Comment (williewillus) [inline src/main/java/net/minecraftforge/event/world/ModifyBlockEvent.java]: wat?

**Comment by williewillus** [comment]
Yes, I missed where you moved it out of PIE, sorry

File: src/main/java/net/minecraftforge/event/world/ModifyBlockEvent.java (hunk 1 above)
Comment (williewillus) [inline src/main/java/net/minecraftforge/event/world/ModifyBlockEvent.java]: did you mean `getWorld().getBlockState(getPos())` by any chance? :P

**Comment by williewillus** [comment]
I still think the majority of scenarios you have brought up are covered by break/place/interact events already, so I don't know what additional use case justifies adding a whole new event

**Comment by williewillus** [comment]
Gravity Gun: Break and place events
Wrenching/dismantling: Break event
Rotating: Break old state, place new state
Etc.

**Comment by LatvianModder** [comment]
Fair enough. I guess then I just have to get modders to use proper events and fire them themselves :P Closing the PR

**Pull request closed without merging**